tracer/contrib/database/sqltraced/utils.go

```diff
@@ -0,0 +1,136 @@
+package sqltraced
+
+import (
+	"database/sql/driver"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/dd-trace-go/tracer"
+	"github.com/go-sql-driver/mysql"
+	"github.com/lib/pq"
+	"golang.org/x/net/context"
+)
+
+// Return all information passed through the DSN:
+func parseDSN(driver interface{}, dsn string) (meta map[string]string, err error) {
+	switch driver.(type) {
+	case *pq.Driver:
+		meta, err = parsePostgresDSN(dsn)
+	case *mysql.MySQLDriver:
+		meta, err = parseMySQLDSN(dsn)
+	}
+	meta = normalize(meta)
+	return
+}
+
+func normalize(meta map[string]string) map[string]string {
+	// Delete the entry "password" for security reasons
+	delete(meta, "password")
```

Comment:
i think we should take the opposite approach here and only allow whitelisted fields. this will ensure we never pluck sensitive data if somebody uses a different field name than password.